catroid/src/androidTest/java/org/catrobat/catroid/test/embroidery/DSTFileGeneratorTest.java

```diff
@@ -0,0 +1,214 @@
+						.MAXDISTANCE);
+				assertTrue(Math.abs(stitchPoints.get(i - 1).point.y - stitchPoints.get(i).point.y) < DSTFileGenerator
+						.MAXDISTANCE);
+			}
+			if (!stitchPoints.get(i).jump) {
+				PointF point = (PointF) points.get(index);
+				assertEquals(stitchPoints.get(i).point.x, point.x * DSTFileGenerator.STITCHPOINTUNITFACTOR);
+				assertEquals(stitchPoints.get(i).point.y, point.y * DSTFileGenerator.STITCHPOINTUNITFACTOR);
+				index += 1;
+			}
+		}
+	}
+
+	@Test
+	public void testHeaderBytes() {
+		final String expectedHeaderString = "LA:testProject    \n" + (char) 0x1A + ("ST:3     \n" + (char) 0x1A
+				+ "CO:1 \n" + (char) 0x1A + "+X:20  \n" + (char) 0x1A + "-X:0   \n" + (char) 0x1A + "+Y:20  \n"
+				+ (char) 0x1A + "-Y:0   \n" + (char) 0x1A + "AX:0    \n" + (char) 0x1A + "AY:0    \n" + (char) 0x1A
+				+ "MX:0    \n" + (char) 0x1A + "MY:0    \n" + (char) 0x1A + "PD:*****\n"
+				+ (char) 0x1A).replace(' ', '\0');
+
+		EmbroideryList points = new EmbroideryList();
+		points.add(new PointF(0, 0));
+		points.add(new PointF(100, 100));
+		points.add(new PointF(0, 0));
+
+		ArrayList<DSTFileGenerator.StitchPoint> stitchPoints = fileGenerator.prepareStitchPoints(points);
+		byte[] header = fileGenerator.getHeaderBytes(stitchPoints);
+		byte[] expectedHeader = expectedHeaderString.getBytes();
+
+		assertEquals(DSTFileGenerator.HEADERMAXBYTE, header.length);
+
+		for (int i = 0; i < DSTFileGenerator.HEADERMAXBYTE; i++) {
+			if (i >= expectedHeader.length) {
+				assertEquals(' ', (char) header[i]);
+			} else {
+				assertEquals(expectedHeader[i], header[i]);
+			}
+		}
+	}
+
+	private PointF getNextPointFromByte(PointF lastPoint, byte b0, byte b1, byte b2) {
+		int xConversionByte = (b0 & 0x3) | ((b1 & 0x3) << 2) | ((b0 & 0xC) << 2) | ((b1 & 0xC) << 4) | ((b2 & 0xC) << 6);
+		int yConversionByte = ((b0 >>> 7) & 0x1) | ((b0 >>> 5) & 0x2) | ((b1 >>> 5) & 0x4) | ((b1 >>> 3) & 0x8)
+				| ((b0 >>> 1) & 0x10) | ((b0 << 1) & 0x20) | ((b1 << 1) & 0x40) | ((b1 << 3) & 0x80)
+				| ((b2 << 6) & 0x100) | ((b2 << 4) & 0x80);
+		int xChange = getValueForByte(xConversionByte);
+		int yChange = getValueForByte(yConversionByte);
+		return new PointF(lastPoint.x + xChange, lastPoint.y + yChange);
+	}
+
+	@Test
+	public void testStitchBytes() {
+		ArrayList<DSTFileGenerator.StitchPoint> stitchpoints = new ArrayList<>();
+		stitchpoints.add(new DSTFileGenerator.StitchPoint(new PointF(0, 0)));
+		stitchpoints.add(new DSTFileGenerator.StitchPoint(new PointF(20, 0)));
+		stitchpoints.add(new DSTFileGenerator.StitchPoint(new PointF(20, 20)));
+		stitchpoints.add(new DSTFileGenerator.StitchPoint(new PointF(0, 20)));
+		stitchpoints.add(new DSTFileGenerator.StitchPoint(new PointF(0, 0)));
+		byte[] stitchBytes = fileGenerator.getStitchesBytes(stitchpoints);
+
+		PointF lastPoint = stitchpoints.get(0).point;
+		int index = 0;
+		for (int i = 0; i < stitchBytes.length; i += 3) {
```

Comment:
please no loops in tests, especially no duplicate indices.
you could also make this an parameterized test that just gets 2 points and does exactly the same.